src/hostapi/oboe/pa_oboe.cpp

```diff
@@ -0,0 +1,1900 @@
+
+    unsigned long framesPerHostCallback;
+    unsigned bytesPerFrame;
+
+    OboeMediator* oboeMediator;
+} OboeStream;
+
+
+class OboeMediator: public AudioStreamCallback{
+public:
+    OboeMediator(OboeStream* i_oboeStream) {
+        m_oboeCallbackStream = i_oboeStream;
+    }
+
+    //Callback function for non-blocking streams
+    DataCallbackResult onAudioReady(AudioStream *audioStream, void *audioData,
+                                    int32_t numFrames) override;
+    //Callback utils
+    void onErrorAfterClose(AudioStream *audioStream, oboe::Result error) override;
+    void resetCallbackCounters();
+    void setOutputCallback() { m_outputBuilder.setDataCallback(this)->setErrorCallback(this); }
+    void setInputCallback() { m_inputBuilder.setDataCallback(this)->setErrorCallback(this); }
+
+    //getter and setter of m_oboeEngine and m_oboeCallbackStream
+    OboeEngine *getEngine() { return m_oboeEngine; }
+    void setEngine(OboeEngine *i_oboeEngine) { m_oboeEngine = i_oboeEngine; }
+
+    OboeStream *getStreamAddress() { return m_oboeCallbackStream; }
+    void setCallbackStream(OboeStream *i_oboeStream) { m_oboeCallbackStream = i_oboeStream; }
+
+    //The only instances of output and input streams that will be used, and their builders
+    std::shared_ptr <AudioStream> m_outputStream;
+    AudioStreamBuilder m_outputBuilder;
+    std::shared_ptr <AudioStream> m_inputStream;
+    AudioStreamBuilder m_inputBuilder;
+
+private:
+    OboeEngine *m_oboeEngine;
+
+    //callback utils
+    OboeStream *m_oboeCallbackStream;
+    unsigned long m_framesProcessed{};
+    PaStreamCallbackTimeInfo m_timeInfo{};
+    struct timespec m_timeSpec{};
+};
+
+
+/**
+ * Stream engine of the host API - Oboe. We allocate only one instance of the engine per OboeStream, and
+ * we call its functions when we want to operate directly on Oboe. More information on each function is
+ * provided right before its implementation.
+ */
+class OboeEngine {
+public:
+    OboeEngine();
+
+    //Stream-managing functions
+    bool tryStream(Direction direction, int32_t sampleRate, int32_t channelCount);
+
+    PaError openStream(OboeStream *oboeStream, Direction direction, int32_t sampleRate,
+                       Usage outputUsage, InputPreset inputPreset);
+
+    bool startStream(OboeStream *oboeStream);
+
+    bool stopStream(OboeStream *oboeStream);
+
+    bool restartStream(OboeStream *oboeStream, int direction);
+
+    bool closeStream(OboeStream *oboeStream);
+
+    bool abortStream(OboeStream *oboeStream);
+
+    //Blocking read/write functions
+    bool writeStream(OboeStream *oboeStream, const void *buffer, int32_t framesToWrite);
+
+    bool readStream(OboeStream *oboeStream, void *buffer, int32_t framesToRead);
+
+    //Engine utils
+    void constructOboeStream(OboeStream* oboeStream);
+
+private:
+    std::shared_ptr <AudioStream> m_testStream;
+    AudioStreamBuilder m_testBuilder;
+
+    OboeMediator* m_terminableMediator;
```

Comment:
The m_ prefix is not used in Android or PortAudio. Can you use mTerminableMediator.